cristin-import/src/main/java/no/unit/nva/cristin/mapper/PublicationInstanceBuilder.java

```diff
@@ -0,0 +1,199 @@
+                .build();
+    }
+
+    private PublicationInstance<? extends Pages> createFeatureArticle() {
+        Range numberOfPages = new Range(extractPagesBegin(), extractPagesEnd());
+        return new FeatureArticle.Builder()
+                .withPages(numberOfPages)
+                .withVolume(extractVolume())
+                .build();
+    }
+
+    private PublicationInstance<? extends Pages> createJournalLetter() {
+        Range numberOfPages = new Range(extractPagesBegin(), extractPagesEnd());
+        return new JournalLetter.Builder()
+                .withPages(numberOfPages)
+                .withVolume(extractVolume())
+                .build();
+    }
+
+    private PublicationInstance<? extends Pages> createJournalReview() {
+        Range numberOfPages = new Range(extractPagesBegin(), extractPagesEnd());
+        return new JournalReview.Builder()
+                .withPages(numberOfPages)
+                .withVolume(extractVolume())
+                .build();
+    }
+
+    private PublicationInstance<? extends Pages> createJournalLeader() {
+        Range numberOfPages = new Range(extractPagesBegin(), extractPagesEnd());
+        return new JournalLeader.Builder()
+                .withPages(numberOfPages)
+                .withVolume(extractVolume())
+                .build();
+    }
+
+    private PublicationInstance<? extends Pages> createJournalCorrigendum() {
+        Range numberOfPages = new Range(extractPagesBegin(), extractPagesEnd());
+        return new JournalCorrigendum.Builder()
+                .withPages(numberOfPages)
+                .withVolume(extractVolume())
+                .build();
+    }
+
+    private PublicationInstance<? extends Pages> createJournalArticle() {
+        Range numberOfPages = new Range(extractPagesBegin(), extractPagesEnd());
+        return new JournalArticle.Builder()
+                .withContent(cristinObject.getSecondaryCategory().toJournalArticleContentType())
+                .withPages(numberOfPages)
+                .withPeerReviewed(HARDCODED_JOURNAL_PEER_REVIEWED)
+                .withVolume(extractVolume())
+                .build();
+    }
+
+    private PublicationInstance<? extends Pages> createReportResearch() {
+        return new ReportResearch.Builder().build();
+    }
+
+    private PublicationInstance<? extends Pages> createDegreePhd() {
+        return new DegreePhd
+                .Builder()
+                .withPages(createMonographPages())
+                .build();
+    }
+
+    private PublicationInstance<? extends Pages> createDegreeMaster() {
+        return new DegreeMaster
+                .Builder()
+                .withPages(createMonographPages())
+                .build();
+    }
+
+    private PublicationInstance<? extends Pages> createChapterArticle() {
+        return new ChapterArticle.Builder().build();
+    }
+
+    private String extractPagesBegin() {
+        return mapper.extractCristinJournalPublication().getPagesBegin();
```

Comment:
A Mapper has a PublicationInstanceBuilder which has a Mapper for just getting a field from the CristinObect?

Reply:
The only reason you are using this method is to handle the case where the cristinObject is null. Require that in the constructor and simplify your code